lib/backend/lite/lite.go

```diff
@@ -791,17 +857,13 @@ func (l *Backend) DeleteRange(ctx context.Context, startKey, endKey backend.Key)
 			return trace.Wrap(err)
 		}
 		defer rows.Close()
-		var keys []backend.Key
 		for rows.Next() {
 			var key backend.Key
 			if err := rows.Scan(&key); err != nil {
 				return trace.Wrap(err)
 			}
-			keys = append(keys, key)
-		}
 
-		for i := range keys {
-			if err := l.deleteInTransaction(l.ctx, keys[i], tx); err != nil {
+			if err := l.deleteInTransaction(l.ctx, key, tx); err != nil {
```

Comment:
I'm pretty sure this is an error or a deadlock - you can't delete in the middle of receiving the output of a query, right? It's weird enough that we're not calling `rows.Close` before starting the delete...

Reply:
The sqlite docs specifically point out that it's fine to delete by rowid after getting a row in a select, but the behavior is sort of undefined if you're modifying the table in other ways while reading it: https://www.sqlite.org/isolation.html#:~:text=No%20Isolation%20Between%20Operations%20On%20The%20Same%20Database%20Connection

Something to keep in mind for future improvements, I suppose - if the `database/sql` driver even lets us do that.